meteor-app/imports/ui/ordering/cart-layout.js

```diff
@@ -9,29 +9,39 @@ Session.set('searchQuery', '')
 export default withTracker((props) => {
   const ordersHandle = Meteor.subscribe('all.orders')
   const removePart = async (OrderId, partId) => {
-    const orderedParts = await Orders.findOne({ _id: OrderId}).orderedParts
-    orderedParts.forEach(part => {
+    const order = await Orders.findOne({ _id: OrderId})
+    order.orderedParts.forEach(part => {
       if(part.partId === partId){
-        Meteor.callAsync('orders.removePart', OrderId, part)
+        order.totalPrice = order.totalPrice - (part.price * part.qty)
+        Meteor.callAsync('order.update', order._id, order)
+        Meteor.callAsync('orders.removePart', order, part )
       }
     })
   }
+  
   const increaseQty = async (orderId, partId) => {
     const order = await Orders.findOne({ _id: orderId})
     order.orderedParts.forEach(part => {
       if(part.partId === partId){
         part.qty += 1
-        Meteor.callAsync('order.updateQty', order._id, order.orderedParts)
+        order.totalPrice += part.price
+        Meteor.callAsync('order.update', order._id, order)
```

Comment:
Again, instead of iterating over data and doing a method call for each part, do it all in one method